src/blockdev.rs

```diff
@@ -61,17 +56,48 @@ fn get_partition_with_label(device: &str, label: &str) -> Result<Option<BlkDev>>
         eprint!("{}", String::from_utf8_lossy(&*result.stderr));
         bail!("lsblk of {} failed", device);
     }
-    let output: LsBlk =
-        serde_json::from_slice(&*result.stdout).chain_err(|| "decoding lsblk JSON")?;
+    let output = String::from_utf8(result.stdout).chain_err(|| "decoding lsblk output")?;
+
+    // walk each device in the output
+    let re = Regex::new("([A-Z-]+)=\"([^\"]+)\"").unwrap();
     let mut found: Option<BlkDev> = None;
-    for dev in output.blockdevices {
-        if dev.label.is_none() || dev.label.as_ref().unwrap() != label {
-            continue;
+    for line in output.lines() {
+        // parse key-value pairs
+        // newer versions of lsblk support JSON but CentOS 7 doesn't
+        let mut fields: HashMap<String, String> = HashMap::new();
+        for cap in re.captures_iter(line) {
+            fields.insert(cap[1].to_string(), cap[2].to_string());
+        }
+
+        // does the label match?
+        match fields.get("LABEL") {
+            None => continue,
+            Some(l) => {
+                if l != label {
+                    continue;
+                }
+            }
         }
+
+        // Older lsblk, e.g. in CentOS 7.6, doesn't support PATH.
+        // Assemble device path from dirname and NAME.
+        let mut path = Path::new(device)
+            .parent()
+            .chain_err(|| format!("couldn't get parent directory of {}", device))?
+            .to_path_buf();
+        match fields.get("NAME") {
+            None => continue,
+            Some(name) => path.push(name),
+        }
+
+        // accept
```

Comment:
Spurious line?

Reply:
Uh, no?

Reply:
Does it spark joy?

On a serious note: :+1:, sorry, it was hanging a bit alone so my brain put it in the garbage pile.